knowledge-base/grid-inline-edit-form.md

```diff
@@ -0,0 +1,227 @@
+                               Icon="@SvgIcon.Cancel">
+                    Cancel
+                </TelerikButton>
+            </FormButtons>
+        </TelerikForm>
+    </DetailTemplate>
+</TelerikGrid>
+
+<style>
+    .no-expand-column .k-hierarchy-col {
+        width: 0;
+    }
+
+    .no-expand-column .k-hierarchy-cell + .k-table-td,
+    .no-expand-column .k-hierarchy-cell + .k-table-th {
+        border-left-width: 0;
+    }
+
+    .no-expand-column .k-hierarchy-cell > * {
+        display: none;
+    }
+
+    .no-expand-column .k-detail-row:hover {
+        background-color: transparent;
+    }
+</style>
+
+@code {
+#nullable enable
+
+    private List<SampleModel> GridData { get; set; } = new();
+
+    private TelerikGrid<SampleModel> GridRef { get; set; } = null!;
+
+    private SampleModel? GridItemInEditMode { get; set; }
+
+    private async Task OnGridEditButtonClick(SampleModel item)
+    {
+        GridItemInEditMode = item.Clone();
+
+        var gridState = GridRef.GetState();
+        gridState.ExpandedItems = new List<SampleModel>() { item };
+
+        await GridRef.SetStateAsync(gridState);
+    }
+
+    private async Task OnGridAddButtonClick()
+    {
+        var newItem = new SampleModel();
+        GridData.Insert(0, newItem);
+        GridRef.Rebind();
+
+        await OnGridEditButtonClick(newItem);
+    }
+
+    private async Task OnGridFormValidSubmit()
+    {
+        if (GridItemInEditMode == null)
+        {
+            return;
+        }
+
+        var originalItemIndex = GridData.FindIndex(x => x.Id == GridItemInEditMode!.Id);
+
+        if (GridItemInEditMode!.Id == 0)
+        {
+            GridItemInEditMode.Id = ++LastId;
+        }
+
+        GridData[originalItemIndex] = GridItemInEditMode!;
+
+        var gridState = GridRef.GetState();
+        gridState.ExpandedItems = new List<SampleModel>();
+
+        await GridRef.SetStateAsync(gridState);
+
+        GridItemInEditMode = null;
+    }
+
+    private async Task OnGridFormCancel()
+    {
+        if (GridItemInEditMode?.Id == 0)
+        {
+            var itemIndex = GridData.FindIndex(x => x.Id == 0);
+            GridData.RemoveAt(itemIndex);
+        }
+
+        var gridState = GridRef.GetState();
+        gridState.ExpandedItems = new List<SampleModel>();
+
+        await GridRef.SetStateAsync(gridState);
+
+        GridItemInEditMode = null;
+    }
+
+    private int LastId { get; set; }
+
+    protected override void OnInitialized()
+    {
+        var rnd = new Random();
+
+        for (int i = 1; i <= 5; i++)
+        {
+            GridData.Add(new SampleModel()
+                {
+                    Id = ++LastId,
+                    Name = $"Name {LastId}",
+                    Price = rnd.Next(1, 100) * 1.23m,
+                    Quantity = rnd.Next(0, 1000),
+                    StartDate = DateTime.Today.AddDays(-rnd.Next(1, 30)).AddMonths(-rnd.Next(1, 100))
+                });
```

Comment:
Does VS for Windows really adds one more tab here? VS Mac doesn't do that.

Reply:
Yes, it was formatted by VS for Windows. I am not sure why.